### PR TITLE
feat: 공지 다이얼로그 백키 반응 추가

### PR DESCRIPTION
**Pull request by yaminsu5674**
- [x]  dialog.tsx 변경
- [x]  공지 클릭 사용처에 pushState

https://www.notion.so/mmyeong/feat-dad41238f5174d8b9fbc33d01366ed37?pvs=4

### CODE REVIEW

**Comment by yaminsu5674** [comment]
* **#163** <a href="https://app.graphite.dev/github/pr/TeamVisioneer/PrayU-Web/163?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @yaminsu5674 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/ui/dialog.tsx [hunk 1]
```diff
@@ -1,10 +1,31 @@
 import * as React from "react";
 import * as DialogPrimitive from "@radix-ui/react-dialog";
 import { X } from "lucide-react";
+import { useEffect } from "react";
 
 import { cn } from "@/lib/utils";
 
-const Dialog = DialogPrimitive.Root;
+const Dialog = ({
```
Comment (treephesians) [inline src/components/ui/dialog.tsx:8]: 이렇게 하는 거군요! 감사합니다 :)

**Review by treephesians — APPROVED** [review]
감사합니다!!

**Review by myeongjunkim — APPROVED** [review]
굳 감사합니다!

**Pull request merged**